plutus-use-cases/hie.project

```diff
@@ -0,0 +1,4 @@
+packages: .
```

Comment:
Not sure what happens if they are both called `cabal.project` I suspect it works more or less as you would expect though (cabal searches up for the first `cabal.project` file and will used that one).

In this case though it will only be used by the [patched version of the hie-bios](https://github.com/mpickering/hie-bios/compare/master...hamishmack:hkm/hie-project-file#diff-d9d44b1f07a4384f5b7934a7591ea3faR429).  So if the hie.yaml file is present in that directory the patched hie-bios will explicitly use that `hie.project` when present.

It and serves two purposes:

* Makes sure just the plutus-uses-cases package treated as a local package.  The object-code branch of ghcide is very slow if there are lots of packages in the `.project` file (I think it likes to build them all).

* Allows us to se the skip plugin option just for ghcide

Reply:
> The object-code branch of ghcide is very slow if there are lots of packages in the .project file (I think it likes to build them all).

:confused: 

Surely it *must* build the dependencies otherwise there's no way it can work?

Reply:
It seems to spend a lot of time figuring out there is nothing to rebuild every time you over over something in VS Code.

Reply:
Ah I see, it works because there's more conditional logic in `shell.nix` :see_no_evil: